programs/gamma/src/migration/raydium/raydium_cp_swap_to_gamma.rs

```diff
@@ -0,0 +1,189 @@
+    #[account(mut)]
+    pub raydium_cp_swap_lp_mint: UncheckedAccount<'info>,
+
+    /// memo program
+    /// CHECK:
+    #[account(address = spl_memo::id())]
+    pub memo_program: UncheckedAccount<'info>,
+
+    /// Owner of the liquidity provided
+    pub gamma_owner: Signer<'info>,
+
+    /// CHECK: pool vault authority
+    #[account(
+        seeds = [
+            crate::AUTH_SEED.as_bytes(),
+        ],
+        bump,
+    )]
+    pub gamma_authority: UncheckedAccount<'info>,
+
+    /// Pool state the owner is depositing into
+    #[account(mut)]
+    pub gamma_pool_state: AccountLoader<'info, PoolState>,
+
+    #[account(
+        mut,
+        seeds = [
+            USER_POOL_LIQUIDITY_SEED.as_bytes(),
+            gamma_pool_state.key().as_ref(),
+            gamma_owner.key().as_ref(), 
+        ],
+        bump,
+    )]
+    pub gamma_user_pool_liquidity: Account<'info, UserPoolLiquidity>,
+
+    // /// CHECK: The destination token account for receive amount_0
+    // pub raydium_recipient_token_account_0: UncheckedAccount<'info>,
+    /// The payer's token account to deposit token_0
+    #[account(
+        mut,
+        token::mint = gamma_token_0_vault.mint,
+        token::authority = gamma_owner
+    )]
+    pub gamma_token_0_account: Box<InterfaceAccount<'info, TokenAccount>>,
+
+
+    // /// CHECK: The destination token account for receive amount_1
+    // pub raydium_recipient_token_account_1: UncheckedAccount<'info>,
+    /// The payer's token account to deposit token_1
+    #[account(
+        mut,
+        token::mint = gamma_token_1_vault.mint,
+        token::authority = gamma_owner
+    )]
+    pub gamma_token_1_account: Box<InterfaceAccount<'info, TokenAccount>>,
+    /// Pool vault for token_0 to deposit into
+    /// The address that holds pool tokens for token_0
+    #[account(
+        mut,
+        constraint = gamma_token_0_vault.key() == gamma_pool_state.load()?.token_0_vault
+    )]
+    pub gamma_token_0_vault: Box<InterfaceAccount<'info, TokenAccount>>,
+    /// Pool vault for token_1 to deposit into
+    /// The address that holds pool tokens for token_1
+    #[account(
+        mut,
+        constraint = gamma_token_1_vault.key() == gamma_pool_state.load()?.token_1_vault
+    )]
+    pub gamma_token_1_vault: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    /// The mint of token_0 vault
+    #[account(
+        address = gamma_token_0_vault.mint
+    )]
+    pub gamma_vault_0_mint: Box<InterfaceAccount<'info, Mint>>,
+
+    /// The mint of token_1 vault
+    #[account(
+        address = gamma_token_1_vault.mint
+    )]
+    pub gamma_vault_1_mint: Box<InterfaceAccount<'info, Mint>>,
+    
+    /// token Program
+    pub token_program: Program<'info, Token>,
+
+    /// Token program 2022
+    pub token_program_2022: Program<'info, Token2022>,
+}
+
+pub fn raydium_cp_swap_to_gamma<'a, 'b, 'c, 'info>(
+    ctx: Context<'a, 'b, 'c, 'info, RaydiumCpSwapToGamma<'info>>,
+    lp_token_amount: u64,
```

Comment:
can we for all instructions also have 2 lp_token_amounts as this depends on lp price.
It is possible that I withdraw 10 lp from radium and then I can invest 15 to gamma.

Reply:
Applies to all instruction,

LP price is based on the trade fee earned so if the fee earned is a significant amount.


We can also calculate the lp token